### DISCUSSION TITLE
Can't Make a Boolean Search Work

### DISCUSSION THREAD

**Discussion by jaredwa** [ravendb/ravendb#16041] [Q&A]
I have a collection with a few million items. Of those, a few hundred thousand have a Boolean field theField. 

When I run a query 

from COLLECTION where theField == true

I get no results even though there should be thousands. I've tried exists(theField) and I also get no results. Am I missing something obvious?

**Comment by ayende** [comment]
That should work, of course.
Can you show the actual query you are running?
Also, is `theField` a boolean or a `"true"` string?

**Reply by jaredwa** [reply]
Sure - broadly:

from 'Posts' where video == true

I've tried a lot of variations

In the object, it's a boolean. This returns no results. 

<img width="1468" alt="Screenshot 2023-03-19 at 2 47 51 PM" src="https://user-images.githubusercontent.com/1483529/226199975-86d7f60c-bc40-4958-89a3-74ba23e0c66f.png">

**Reply by garayx** [reply]
can you share a sample document that reproduce the issue and the exact query you are using ?

**Reply by jaredwa** [reply]
Oh! It started working. I think it was a bad index? 

from 'Posts' where video == true

Just... works all of a sudden. That's SO Raven.

**Reply by garayx** [reply]
Queries on static index will start have `index` keyword in the query, ie: `from index "Orders/Totals"` 
Your query should create auto index (https://ravendb.net/docs/article-page/5.4/csharp/indexes/creating-and-deploying#auto-indexes), so maybe the created auto index didn't finish to fully index the data set, and that's why you didn't have all results.

**Reply by jaredwa** [reply]
I'm starting to realize that a lot of my query failed are just the database's index being stale. Which is like... how hard would it be to put a more explicit error to that? Like, "Literally this search won't work as expected because there's a query you didn't initiate that happens to be rebuilding and also I'm not going to tell you when and also I'mma charge you for all the space."

Something like that. Thanks, though.

**Reply by ayende** [reply]
You can do that by putting this as part of your document store init:

```
store.OnBeforeQuery += (sender, q) => q.QueryCustomization.WaitForNonStaleResults();
```

Note that this is not the default for a reason, since it means that when most of the time you don't care for the latest updates, you are faster.

**Reply by jaredwa** [reply]
Thanks! But these often take hours to rebuild - that's... would just be blocker code. I meant why not spit out an error or a warning?

**Reply by ayende** [reply]
You can do that too, we are providing you that information.
The question is, how do you expect to _get_ that warning?

**Reply by ayende** [reply]
I'm not sure that I follow.
*why* would that take hours to build? 
If you generate a new index each time, sure (if the db is big enough)
But for production, you usually already have all the idnexes.
